pkg/pillar/nireconciler/linux_config.go

```diff
@@ -336,6 +342,80 @@ func (r *LinuxNIReconciler) getIntendedGlobalState() dg.Graph {
 	return intendedCfg
 }
 
+func (r *LinuxNIReconciler) getIntendedHostSysctl() linux.Sysctl {
+	var bridgeCallIptables, bridgeCallIp6tables bool
+	// If ACLs allow everything without rate-limiting and flow-logging
+	// is not enabled in any network instance, we can disable calling
+	// iptables from bridges to improve performance of L2-forwarding.
+	allIpv4TrafficAllowed := true
+	allIpv6TrafficAllowed := true
+	flowLoggingIsEnabled := false
+	for _, ni := range r.nis {
+		if ni.deleted {
+			continue
+		}
+		if ni.config.EnableFlowlog {
+			flowLoggingIsEnabled = true
+			break
+		}
+	}
+	for _, app := range r.apps {
+		if app.deleted {
+			continue
+		}
+		for _, adapter := range app.config.AppNetAdapterList {
+			var hasAllowAllRuleIPv4, hasAllowAllRuleIPv6 bool
+			var hasLimitRule, hasDropRule bool
+			for _, ace := range adapter.ACLs {
+				// Default action (if not specified) is to allow traffic to continue.
+				allowRule := true
+				for _, action := range ace.Actions {
+					if action.Drop {
+						hasDropRule = true
```

Comment:
Isn't it enough to mark that iptables are necessary in this case? I'm still trying to figure out why we should continue with the other checks after that.

Reply:
I have rewrote and split the function to avoid unnecessary checks.